lib/messages/chainlock.js

```diff
@@ -0,0 +1,276 @@
+ * @throws {TypeError} - If the argument was not recognized
+ * @private
+ */
+ChainLock._from = function _from(arg) {
+  var info = {};
+  if (BufferUtil.isBuffer(arg)) {
+    info = ChainLock._fromBufferReader(BufferReader(arg));
+  } else if (isObject(arg)) {
+    info = ChainLock._fromObject(arg);
+  } else if (isHexString(arg)) {
+    info = ChainLock.fromHex(arg);
+  } else {
+    throw new TypeError('Unrecognized argument for ChainLock');
+  }
+  return info;
+}
+
+ChainLock._fromObject = function _fromObject(data) {
+  $.checkArgument(data, 'data is required');
+  var blockHash = data.blockHash || data.blockhash;
+  var signature = data.signature;
+  if (isString(blockHash)) {
+    blockHash = BufferUtil.reverse(Buffer.from(blockHash, 'hex'));
+  }
+
+  if (isString(data.signature)) {
+    signature = Buffer.from(data.signature, 'hex');
+  }
+  var info = {
+    height: data.height,
+    blockHash: blockHash,
+    signature: signature
+  }
+  return info;
+}
+
+/**
+ * @param {BufferReader} br - Chainlock data
+ * @returns {Object} - An object representing the chainlock data
+ * @private
+ */
+ChainLock._fromBufferReader = function _fromBufferReader(br) {
+  var info = {};
+  info.height = br.readInt32LE();
+  info.blockHash = br.read(SHA256_HASH_SIZE);
+  info.signature = br.read(BLS_SIGNATURE_SIZE);
+  return info;
+};
+
+/**
+ * @param {BufferReader} br A buffer reader of the block
+ * @returns {ChainLock} - An instance of ChainLock
+ */
+ChainLock.fromBufferReader = function fromBufferReader(br) {
+  $.checkArgument(br, 'br is required');
+  var data = ChainLock._fromBufferReader(br);
+  return new ChainLock(data);
+};
+
+/**
+ * Creates ChainLock from a hex string.
+ * @param {String} string - A hex string representation of the chainLock
+ * @return {ChainLock} - An instance of ChainLock
+ */
+ChainLock.fromHex = ChainLock.fromString = function fromString(string) {
+  return ChainLock.fromBuffer(Buffer.from(string, 'hex'));
+};
+
+
+/**
+ * Creates ChainLock from a Buffer.
+ * @param {Buffer} buffer - A buffer of the chainLock
+ * @return {ChainLock} - An instance of ChainLock
+ */
+ChainLock.fromBuffer = function fromBuffer(buffer) {
+  return ChainLock.fromBufferReader(new BufferReader(buffer));
+};
+
+/**
+ * Create ChainLock from an object
+ * @param {Object} obj - an object with all properties of chainlock
+ * @return {ChainLock}
+ */
+ChainLock.fromObject = function fromObject(obj) {
+  var data = ChainLock._fromObject(obj);
+  return new ChainLock(data);
+};
+
+/**
+ * Verify that the signature is valid against the Quorum using quorumPublicKey
+ * @param {QuorumEntry} quorumEntry - quorum entry to test signature against
+ * @returns {Promise<Boolean>} - if the signature is valid for this quorum
+ */
+ChainLock.prototype.verifySignatureAgainstQuorum = async function verifySignatureAgainstQuorum(quorumEntry) {
+  var self = this;
+  return new Promise(function(resolve) {
```

Comment:
Why not use `await` here instead of using promises?

Reply:
Actually, even async is an error here... (very old version that we keep supporting for some reason)

Reply:
I think we should move on from it. The plan was to slowly introduce a new feature to new files and refactor the files we modify during the code. I wonder what's @shuplenkov @shumkov @jawid-h take on this

Reply:
I agree we need to move on from this. 
I'm willing to do the 13 task to do it, but the introducing slowly I don't know yet.

Reply:
We've started the migration process towards a new standard 2 years ago, and no work has been done so far. It seems like if we don't do it gradually we just will never do it, and introducing new features to the lib will just complicate the process over time, as they just can be written once and we don't need to rewrite them later. If we don't start now, it will just be in limbo forever, and I really don't want this to happen

Reply:
This PR is a great opportunity to do so - it's a "major" version bump, and this is also completely new code that won't be used by any older app

Reply:
Can we consider 0.19 as a major version bump since we introduce breaking changes with minor versions, atm?